tests/object_explorer/test_routing.py

```diff
@@ -93,28 +95,38 @@ def test_routing_target_get_nodes_not_empty(self):
         self.assertEqual(output[1].node_type, 'Folder')
         self.assertIs(output[2], node1)
         self.assertIs(output[3], node2)
-
+        
         # ... The node generator should have been called
-        node_generator.assert_called_once_with(False, current_path, session, match_params)
+        node_generator.assert_called_once_with(False, current_path, object_explorer_session, match_params)
 
     # ROUTING TABLE TESTS ##################################################
     def test_routing_table(self):
         # Make sure that all keys in the routing table are regular expressions
         # Make sure that all items in the routing table are RoutingTargets
         re_class = re.compile('^/$').__class__
-        for key, item in routing.ROUTING_TABLE.items():
+        for key, item in PG_ROUTING_TABLE.items():
             self.assertIsInstance(key, re_class)
-            self.assertIsInstance(item, routing.RoutingTarget)
+            self.assertIsInstance(item, session.RoutingTarget)
 
     def test_routing_invalid_path(self):
         # If: Ask to route a path without a route
         # Then: I should get an exception
+        service_provider =  ServiceProvider(None, {}, PG_PROVIDER_NAME)
+        object_explorer_service = ObjectExplorerService()
+        object_explorer_service.service_provider = service_provider
+        object_explorer_service._routing_table = PG_ROUTING_TABLE
+
         with self.assertRaises(ValueError):
-            routing.route_request(False, ObjectExplorerSession('session_id', ConnectionDetails()), '!/invalid!')
+            object_explorer_service._route_request(False, 
+                ObjectExplorerSession('session_id', ConnectionDetails()), '!/invalid!/')
 
     def test_routing_match(self):
         # If: Ask to route a request that is valid
-        output = routing.route_request(False, ObjectExplorerSession('session_id', ConnectionDetails()), '/')
+        service_provider =  ServiceProvider(None, {}, PG_PROVIDER_NAME)
+        object_explorer_service = ObjectExplorerService()
+        object_explorer_service.service_provider = service_provider
```

Comment:
As line 125- 128 is common with line 114-117, we should use setup method and define them there and just call in both the tests

Reply:
I found that the setUp method is a called before every test. Would it be ok to define a setUp method that initializes an ObjectExplorerService object, even though only 2 of these tests would use this object? or did you mean to create any helper method, not necessarily setUp?

Reply:
yes, its okay if we add it only for 2 tests now. We might add more tests later. It's good if we can put the redundant stuff in setUp.